python/tests/integration/connectors/memory/test_postgres.py

```diff
@@ -1,214 +1,206 @@
+        assert result_before_upsert is None
 
-@pytest.mark.asyncio
-async def test_get_nearest_match(connection_string, memory_record1, memory_record2):
-    memory = PostgresMemoryStore(connection_string, 2, 1, 5)
-    try:
-        await memory.create_collection("test_collection")
-        await memory.upsert_batch("test_collection", [memory_record1, memory_record2])
-        test_embedding = memory_record1.embedding.copy()
-        test_embedding[0] = test_embedding[0] + 0.01
-
-        result = await memory.get_nearest_match(
-            "test_collection", test_embedding, min_relevance_score=0.0, with_embedding=True
-        )
+        await collection.upsert(df)
+        result: pd.DataFrame = await collection.get(1)
         assert result is not None
-        assert result[0]._id == memory_record1._id
-        assert result[0]._text == memory_record1._text
-        assert result[0]._timestamp == memory_record1._timestamp
-        for i in range(len(result[0]._embedding)):
-            assert result[0]._embedding[i] == memory_record1._embedding[i]
-    except PoolTimeout:
-        pytest.skip("PoolTimeout exception raised, skipping test.")
+        row = result.iloc[0]
+        assert row.id == record.id
+        assert row.embedding == record.embedding
+        assert row.data == record.data
+
+        await collection.delete(1)
+        result_after_delete = await collection.get(1)
+        assert result_after_delete is None
+    finally:
+        await collection.delete_collection()
 
 
 @pytest.mark.asyncio
-async def test_get_nearest_matches(connection_string, memory_record1, memory_record2, memory_record3):
-    memory = PostgresMemoryStore(connection_string, 2, 1, 5)
-    try:
-        await memory.create_collection("test_collection")
-        await memory.upsert_batch("test_collection", [memory_record1, memory_record2, memory_record3])
-        test_embedding = memory_record2.embedding
-        test_embedding[0] = test_embedding[0] + 0.025
-
-        result = await memory.get_nearest_matches(
-            "test_collection",
-            test_embedding,
-            limit=2,
-            min_relevance_score=0.0,
-            with_embeddings=True,
-        )
-        assert len(result) == 2
-        assert result[0][0]._id in [memory_record3._id, memory_record2._id]
-        assert result[1][0]._id in [memory_record3._id, memory_record2._id]
-    except PoolTimeout:
-        pytest.skip("PoolTimeout exception raised, skipping test.")
+async def test_upsert_get_and_delete_batch(simple_collection: VectorStoreRecordCollection):
+    record1 = SimpleDataModel(id=1, embedding=[1.1, 2.2, 3.3], data={"key": "value"})
+    record2 = SimpleDataModel(id=2, embedding=[4.4, 5.5, 6.6], data={"key": "value"})
+
+    result_before_upsert = await simple_collection.get_batch([1, 2])
+    assert result_before_upsert is None
+
+    await simple_collection.upsert_batch([record1, record2])
+    result = await simple_collection.get_batch([1, 2, 3])
```

Comment:
Is this intentional?
```suggestion
    result = await simple_collection.get_batch([1, 2])
```

Reply:
This was intentional. I got tripped up on the signature of `get_batch`, in that it returns only the keys that exist in the store and doesn't return anything around keys that didn't exist. The signature for `get_batch` could have been with a return value of `list[TModel | None]` rather than `OneOrMany[TModel] | None`, which I initially expected. I kept the inclusion of a non-existent key in the test to prove out the correct behavior, which I believe is to only return TModel records for existing keys and ignore non-existing keys.

I'll add a comment to explain that for readability.